readthedocs/search/parsers.py

```diff
@@ -424,137 +476,45 @@ def _process_fjson(self, fjson_path):
-        nodes_to_be_removed += body.css('.toctree-wrapper') + body.css('.contents.local.topic')
+        # TODO: see if we really need to remove TOC elements like below?
+        # benjaoming: I didn't see this in sphinx-rtd-theme, however since it wraps the menu in
+        # a <nav>, it's already covered. I didn't see this match local table of contents, neither
+        # and they are also wrapped in a <nav> so covered by _clean_body as well.
+        nodes_to_be_removed = itertools.chain(
+            body.css(".toctree-wrapper"),
+            body.css(".contents.local.topic"),
+        )
 
         # removing all nodes in list
         for node in nodes_to_be_removed:
             node.decompose()
 
         return body
 
-    def _generate_domains_data(self, body):
-        """
-        Generate sphinx domain objects' docstrings.
-
-        Returns a dict with the generated data.
-        The returned dict is in the following form::
-
-            {
-                "domain-id-1": "docstrings for the domain-id-1",
-                "domain-id-2": "docstrings for the domain-id-2",
-            }
-
-        .. note::
-
-           Only the first `self.max_inner_documents` domains are returned.
-        """
-
-        domain_data = {}
-        dl_tags = self._get_sphinx_domains(body)
-        number_of_domains = 0
-
-        for dl_tag in dl_tags:
-
-            dt = dl_tag.css('dt')
-            dd = dl_tag.css('dd')
-
-            # len(dt) should be equal to len(dd)
-            # because these tags go together.
-            for title, desc in zip(dt, dd):
-                try:
-                    id_ = title.attributes.get('id', '')
-                    if id_:
-                        # Create a copy of the node,
-                        # since _parse_domain_tag will modify it.
-                        copy_desc = HTMLParser(desc.html).body.child
-                        docstrings = self._parse_domain_tag(copy_desc)
-                        domain_data[id_] = docstrings
-                        number_of_domains += 1
-                    if number_of_domains >= self.max_inner_documents:
-                        log.warning(
-                            'Limit of inner domains exceeded.',
-                            project_slug=self.project.slug,
-                            version_slug=self.version.slug,
-                            limit=self.max_inner_documents,
-                        )
-                        break
-                except Exception:
-                    log.exception('Error parsing docstring for domains')
-
-        return domain_data
-
-    def _parse_domain_tag(self, tag):
-        """Returns the text from the description tag of the domain."""
-
-        # Remove all nested domains,
-        # since they are already parsed separately.
-        nested_domains = self._get_sphinx_domains(tag)
-        for node in nested_domains:
-            if tag != node:
-                node.decompose()
```

Comment:
Looks like this is where our current code handled removing the duplicate nested nodes, so we do need to keep this behavior in the new implementation. This is, index the content from nested description lists just once.

Reply:
yes, it has been moved, I mentioned you now in a comment where it's now located.